modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -500,16 +493,9 @@ private CPDefinition _addOrUpdateProduct(Product product) throws Exception {
 		Category[] categories = product.getCategories();
 
 		if (categories != null) {
-			List<Long> assetCategoryIds = new ArrayList<>();
-
-			for (Category category : categories) {
-				if (category.getId() != null) {
-					assetCategoryIds.add(category.getId());
-				}
-			}
-
 			serviceContext.setAssetCategoryIds(
-				ArrayUtil.toLongArray(assetCategoryIds));
+				transformToLongArray(
```

Comment:
@JanisZhang We are going to revert the change that adds this method to the base classes in RestBuilder. Please resend without this change.

Reply:
@dantewang OK.